### DISCUSSION TITLE
Install using sys on a single HDD

### DISCUSSION THREAD

**Discussion by floriangit** [jbruchon/elks#1597] [Q&A] [answered]
Hi all,

I got a 286 AMD vintage PC with a working VGA adapter and 1x 1.44MB floppy as well as 1x HDD with 40MB or so.
It runs DOS5/Win3.0 and of course I backed up everything and went straight to try ELKS :-) so far only on using a "Live-Floppy", lol.

Now I'm a bit hesitant after booting and reading around. The HDD has 4 DOS partitions but it seems the current DOS system is taking all space of /dev/hda1. hda2, hda3, hda4 show sizes around 2 million, which looks suspicious to sizeof(uint) on 16bit?

Anyhow my maybe newbie question is: how would you tackle the installation?
1) Is shrinking supported? Maybe I should not ask this for a 30 year old software, but maybe some genius has experience?
2) If not possible: I would fdisk /dev/hda, delete the partition, write/quite, reboot and fdisk again to create a new DOS partition for elks (sys) to use?
3) In case of (2): would you prefer MINIX-FS or FAT - any is "better"? I understand from reading that filesystems are not really recoverable, like modern ext4 etc. Is that true? Maybe I should be more worried about using a spinning HDD which is 35yrs old and still working! :-)

Thanks!

**Comment by ghaerr** [comment, marked as answer]
Hello @floriangit,

The ELKS project itself doesn't have any utilities for shrinking or expanding any already-in-use partitions and saving the partition data. You'd be better off to use a professional program for DOS that has been well tested if you want to go that route.

With regards to MINIX vs FAT - for ELKS, MINIX is much faster. But, for now, I would recommend that you leave the partitions exactly as they are, and attempt to mount any of the partitions after booting from floppy, to check that they work OK. ELKS should be able to handle that no problem using something like `mount /dev/hda1 /mnt`, etc.

After you've done that and played around a bit, a decision can be made that allows for making a one-time conversion to MINIX. Keep in mind though, that although ELKS seems to work well, not all features are implemented for FAT filesystem - for instance the secondary FAT table backup is not maintained. That means that you may or may not want to use the disk for DOS afterwards for serious purposes.

None of the ELKS MINIX (or FAT) filesystems are recoverable: they're not journaled and developed way before that became popular. I think our MINIX v1 format came into being during MINIX development in the 1980s or early '90s.

> Maybe I should be more worried about using a spinning HDD which is 35yrs old and still working! :-)

Yes, I would definitely be worried about that!

Thank you!

**Reply by floriangit** [reply]
Thanks for the helpful and quick reply.

> I would recommend that you leave the partitions exactly as they are, and attempt to mount any of the partitions after booting from floppy, to check that they work OK. ELKS should be able to handle that no problem using something like mount /dev/hda1 /mnt, etc.

It does so far. No problems with mount, fdisk du, df, etc. on the legacy DOS/Win partition.

> a decision can be made that allows for making a one-time conversion to MINIX

Not well understood what you mean with a conversion? Could you elaborate? I suggested to delete DOS/Win with fdisk (since you also mentioned that shrinking is not an option) and I guess, that's the only viable option then to delete and start with ELKS on the HDD, no? Other options?

> That means that you may or may not want to use the disk for DOS afterwards for serious purposes.

Good point and an important one. In case things go not in the way I wanted, would that imply it's maybe impossible to just boot with my M$-DOS disks again and make everything DOS again? Or FreeDOS etc.? I don't have any sensible data on the machine, except the DOS games which I backed-up on floppy disks.

Thanks.

**Reply by ghaerr** [reply]
Glad to hear everything is working on the FAT partitions!

So, you may want to continue playing and "convert" a partition to MINIX.

I guess I didn't mean a "conversion" exactly, just that you would reformat the FAT filesystem as MINIX, thus converting its type. All data would be lost. Should you want to play with this, the ELKS `mkfs` command can be used from within ELKS to format a partition to MINIX format. (You don't need to re-`fdisk` the partition if you don't want to, and ELKS ignores the partition type).

You can always reformat the HD all over again as long as you keep around an MSDOS or FreeDOS boot floppy.

**Reply by floriangit** [reply]
Thank you.
I successfully mkfs'ed /dev/hda1 now and it's a MINIX FS. It shows empty as expected. Bye Microsoft.
Now on to the _sys_ util... # sys /dev/hda1

**Error: Target and System filesystem must be the same type**

Comes down to makeboot.c:454

I will keep poking, but in this case: how would the _sys_ and _makeboot_ utils be of any use? Is it expected to create a MINIX floppy for booting? Since in the docs I read to start with the FAT one: fd1440.img. 
I'll try the minix one from image folder, but maybe consider to update the docs? :-)

**Reply by floriangit** [reply]
got it running with the minix fd image!

Since I saw some discussions re. performance: I did not measure but booting the Minx 1440fd felt 30-40% faster than the DOS 1440fd ....

**Reply by tyama501** [reply]
Hello @ghaerr ,

Is there still under 32 MBytes limitation for minixfs?

**Reply by ghaerr** [reply]
Hello @tyama501,

No, MINIX filesystems can be as large as 65 MBytes, that was fixed a while ago.

**Reply by ghaerr** [reply]
Hello @floriangit,

I'm glad to hear you were able to get a MINIX filesystem up and running on your HDD. Sorry about `sys`, I'm aware of the original problem of having to match the source and destination filesystem types... did you end up solving this by generating a MINIX boot floppy to run `sys` from?

**Reply by floriangit** [reply]
> did you end up solving this by generating a MINIX boot floppy to run sys from?

Yes, thanks, that did the trick. ELKS is now running just fine! :)

> MINIX filesystems can be as large as 65 MBytes

Yes, today I used a 40000 (~40MB) size with _mkfs_ which defaulted to MINIX FS.
